### PR TITLE
[Synthetics] Add support for global variables for synthetics

### LINKED ISSUE #660: Feature request: Global variables for synthetics
Datadog synthetic tests support the use of global variables. One of the use cases for this, would be a shared secret that can be used by multiple Synthetic tests, that will be shared with for example a centralized AWS authorizer.

This provider (to my knowledge), doesn't have a definition for this:
https://docs.datadoghq.com/synthetics/settings/?tab=specifyvalue#global-variables

This could be used in combination with for example SSM/parameter store, or with the Secret Manager: 
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/secretsmanager_secret_version

### PR DESCRIPTION
**Pull request by shibataka000**
## Description
Add support for global variables for synthetics. Close #660 .

## Tests
```bash
$ export DD_API_KEY="..."
$ export DD_APP_KEY="..."

$ make cassettes TESTARGS="-run TestAccDatadogSyntheticsBrowserTest"
cd `mktemp -d`; GO111MODULE=auto GOFLAGS='' go get -u gotest.tools/gotestsum; cd -
/home/takao/src/terraform-provider-datadog
==> Checking that code complies with gofmt requirements...
RECORD=true TF_ACC=1 gotestsum --format testname -- $(go list ./... |grep -v 'vendor') -v -run TestAccDatadogSyntheticsBrowserTest -timeout 120m
EMPTY .
PASS datadog.TestAccDatadogSyntheticsBrowserTest_Basic (4.82s)
PASS datadog.TestAccDatadogSyntheticsBrowserTest_importBasic (5.36s)
PASS datadog.TestAccDatadogSyntheticsBrowserTest_Updated (6.30s)
2020/10/08 20:02:00 Datadog Tracer v1.25.0 ERROR: lost 9 traces: Post "http://localhost:8126/v0.4/traces": dial tcp 127.0.0.1:8126: connect: connection refused, 3 additional messages skipped (first occurrence: 08 Oct 20 20:01 JST)
PASS datadog
EMPTY version

DONE 3 tests in 8.384s

$ make testacc TESTARGS="-run TestAccDatadogSyntheticsBrowserTest"
cd `mktemp -d`; GO111MODULE=auto GOFLAGS='' go get -u gotest.tools/gotestsum; cd -
/home/takao/src/terraform-provider-datadog
==> Checking that code complies with gofmt requirements...
TF_ACC=1 gotestsum --format testname -- $(go list ./... |grep -v 'vendor') -v -run TestAccDatadogSyntheticsBrowserTest -timeout 120m
EMPTY .
PASS datadog.TestAccDatadogSyntheticsBrowserTest_Basic (3.18s)
PASS datadog.TestAccDatadogSyntheticsBrowserTest_importBasic (3.29s)
PASS datadog.TestAccDatadogSyntheticsBrowserTest_Updated (5.56s)
2020/10/08 12:05:34 Datadog Tracer v1.25.0 ERROR: lost 13 traces: Post "http://localhost:8126/v0.4/traces": dial tcp 127.0.0.1:8126: connect: connection refused, 2 additional messages skipped (first occurrence: 08 Oct 20 12:05 JST)
PASS datadog (cached)
EMPTY version

DONE 3 tests in 0.556s
```

### CODE REVIEW

File: datadog/resource_datadog_synthetics_test_.go [hunk 1]
```diff
@@ -126,6 +127,35 @@ func resourceDatadogSyntheticsTest() *schema.Resource {
 					},
 				},
 			},
+			"variables": {
```
Comment (romainberger) [inline datadog/resource_datadog_synthetics_test_.go:130]: For lists like that we tend to use the singular name, so `variable` here

File: datadog/resource_datadog_synthetics_test_.go [hunk 2]
```diff
@@ -126,6 +127,35 @@ func resourceDatadogSyntheticsTest() *schema.Resource {
...
+					Schema: map[string]*schema.Schema{
+						"example": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
```
Comment (romainberger) [inline datadog/resource_datadog_synthetics_test_.go:143]: The format of a variable's name needs to follow a pattern, exactly the same as this one: https://github.com/DataDog/terraform-provider-datadog/blob/master/datadog/resource_datadog_synthetics_global_variable.go#L24

File: docs/resources/synthetics_test.md [hunk 3]
```diff
@@ -239,6 +259,12 @@ The following arguments are supported:
   - `params`: (Required) Parameters for the step as JSON string.
   - `allow_failure`: (Optional) Determines if the step should be allowed to fail.
   - `timeout`: (Optional) Used to override the default timeout of a step.
+- `variables`: (Optional) Array of variables used for the test.
+  - `type`: (Required) Type of browser test variable. Allowed enum values: "element","email","global","text"
+  - `name`: (Required) Name of the variable.
+  - `example`: (Optional) Example for the variable.
+  - `id`: (Optional) ID for the variable.
```
Comment (romainberger) [inline docs/resources/synthetics_test.md:266]: This is actually only used (and required) in the case of using a variable of type `global`. In this case it is the ID of the global variable to use.

Comment (shibataka000) [reply docs/resources/synthetics_test.md]: Thank you for your comment. I update description.

**Review by romainberger — APPROVED** [review]
Looks good to me!

**Review by therve — APPROVED** [review]

**Pull request merged**